src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,152 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> firstNode;
+    private Node<T> lastNode;
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(null, value, null);
+        if (size == 0) {
+            firstNode = newNode;
+        } else {
+            newNode.prev = lastNode;
+            lastNode.next = newNode;
+        }
+        lastNode = newNode;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+        } else if (index == 0) {
+            Node<T> newNode = new Node<>(null, value, firstNode);
+            firstNode.prev = newNode;
+            firstNode = newNode;
+            size++;
+        } else {
+            addNodeInside(value, index);
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T value : list) {
+            add(value);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        return findNodeByIndex(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        Node<T> current = findNodeByIndex(index);
+        T old = current.value;
+        current.value = value;
+        return old;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        Node<T> nodeToRemove = findNodeByIndex(index);
+        unlink(nodeToRemove);
+
+        return nodeToRemove.value;
     }
 
     @Override
     public boolean remove(T object) {
-        return false;
+        Node<T> removedNode = findByElement(object);
+        if (removedNode == null) {
+            return false;
+        }
+        unlink(removedNode);
+        return true;
```

Comment:
here it would be actually less code if you moved findByElement logic here, call unlink inside of if + return true, false outside of the loop